backend/app/testbot/server.py

```
@@ -3,16 +3,124 @@
+        # Update conversation history with the formatted response
+        client_history.append({"query": query, "response": raw_response})
+        # Keep only last 5 exchanges to manage context size
+        conversation_history[client_id] = client_history[-5:] if len(client_history) > 5 else client_history
+        
+        # Return the appropriate response format
+        if is_json_response and product_data:
+            return product_data       
+        else:
+            return json_response
+    
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)